### PR TITLE
Caching: Add an option to cache results in memcache

### LINKED ISSUE #10: Caching: add a memcache plugin
Allow blessyou (since its deterministic) to use memcache for memoization.

### PR DESCRIPTION
**Pull request by danielbeardsley**
Cache key is built from hash of less code + options + sesssionid.

Often compiling less takes ~5-50ms. Since less files just don't change
much and the result is deterministict, caching helps significantly
here.

I added a test that mocks the underlying memcache.

Closes #10

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -101,16 +102,19 @@ function parseLess(req) {
...
       res.writeHead(200, "Content-Type: text/css");
       res.end(css);
-      return Q.fcall(function(){
-         return css;
-      });
+      return css;
    }
 }
 
+function renderCss(req, res) {
```
Comment (djmetzle) [inline server.js]: Maybe order this definition above `outputCss` to reflect the usage ordering? No biggie though. Might even be outdated...

File: cache.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+         })
+      }
+   }
+}
+
+function getCacheKey(session, lessCode, parseOptions) {
+   return md5([lessCode, JSON.stringify(parseOptions) || '']);
+}
+
+function md5(strings) {
```
Comment (djmetzle) [inline cache.js]: This function already exists in a slightly different form:
https://github.com/iFixit/blessyou/blob/4d915da7e5e868f80571f74a1688fcf6c4ba7892/sessions.js#L62-L67

It looks like you could just map over the array in `getCacheKey`?

**Comment by djmetzle** [comment]
CR :+1: Seems legit.

**Comment by danielbeardsley** [comment]
Passes CI!

File: cache.js (hunk 2 above)
Comment (danielbeardsley) [reply cache.js]: I was going by the rule of three. This is only the second repetition and it's only a few lines (and it's different).

Also, the other function would end up using this one cause this creates a hash of concatenated strings.

File: .travis.yml [hunk 3]
```diff
@@ -1,4 +1,4 @@
 language: node_js
 node_js:
-  - "0.8"
-  - "0.10"
+  - "6"
```
Comment (djmetzle) [inline .travis.yml]: "modern node"

**Comment by djmetzle** [comment]
More CR :package:

File: server.js [hunk 4]
```diff
@@ -88,6 +88,7 @@ function handleFailure(req, res) {
    return function (err) {
       var message = less.formatError(err, {color:false})
       l("Got Error: " + req.url + "\n" +  message)
+      l(err);
```
Comment (jarstelfox) [inline server.js]: I would suggest making [console.trace](https://www.google.com/search?q=console.trace&rlz=1C5CHFA_enUS700US702&oq=console.trace&aqs=chrome..69i57j0l5.925j0j1&sourceid=chrome&ie=UTF-8) This way you always get the trace

**Comment by jarstelfox** [comment]
CR 🌵

**Pull request merged**